src/components/swap/ConfirmSwapModal/index.tsx

```diff
@@ -1,10 +1,126 @@
 import { Trans } from '@lingui/macro'
+import { Percent } from '@uniswap/sdk-core'
+import { useCallback, useMemo } from 'react'
 
-import ConfirmSwapModalMod from './ConfirmSwapModalMod'
 import TradeGp from 'state/swap/TradeGp'
 import { TokenAmount } from '@cow/common/pure/TokenAmount'
+import TransactionConfirmationModal, {
+  ConfirmationModalContent,
+  OperationType,
+  TransactionErrorContent,
+} from 'components/TransactionConfirmationModal'
+import { SwapModalFooter } from 'components/swap/SwapModalFooter'
+import SwapModalHeader from 'components/swap/SwapModalHeader'
+import { useWalletDetails } from '@cow/modules/wallet'
+import { SwapConfirmState } from '@cow/modules/swap/state/swapConfirmAtom'
+import { RateInfoParams } from '@cow/common/pure/RateInfo'
+import { useIsSafeApprovalBundle } from '@cow/modules/limitOrders/hooks/useIsSafeApprovalBundle'
+import { TokenSymbol } from '@cow/common/pure/TokenSymbol'
 
-export * from './ConfirmSwapModalMod'
+type ConfirmSwapModalProps = {
+  swapConfirmState: SwapConfirmState
+  trade: TradeGp | undefined
+  recipient: string | null
+  priceImpact?: Percent
+  allowedSlippage: Percent
+  onAcceptChanges: () => void
+  onConfirm: () => void
+  onDismiss: () => void
+  rateInfoParams: RateInfoParams
+}
+
+export function ConfirmSwapModal({
+  trade,
+  swapConfirmState,
+  onAcceptChanges,
+  allowedSlippage,
+  onConfirm,
+  onDismiss,
+  recipient,
+  priceImpact,
+  rateInfoParams,
+}: ConfirmSwapModalProps) {
+  const { swapErrorMessage, showConfirm, attemptingTxn, txHash, tradeToConfirm: originalTrade } = swapConfirmState
+  const { allowsOffchainSigning } = useWalletDetails()
+  const showAcceptChanges = useMemo(
+    () => Boolean(trade && originalTrade && tradeMeaningfullyDiffers(trade, originalTrade)),
+    [originalTrade, trade]
+  )
+
+  const modalHeader = useCallback(() => {
+    return trade ? (
+      <SwapModalHeader
+        trade={trade}
+        rateInfoParams={rateInfoParams}
+        allowsOffchainSigning={allowsOffchainSigning}
+        allowedSlippage={allowedSlippage}
+        priceImpact={priceImpact}
+        recipient={recipient}
+        showAcceptChanges={showAcceptChanges}
+        onAcceptChanges={onAcceptChanges}
+      />
+    ) : null
+  }, [
+    trade,
+    allowsOffchainSigning,
+    allowedSlippage,
+    priceImpact,
+    recipient,
+    showAcceptChanges,
+    onAcceptChanges,
+    rateInfoParams,
+  ])
+
+  const isSafeApprovalBundle = useIsSafeApprovalBundle(trade?.inputAmount)
+  const buttonText = useMemo(
+    () =>
+      isSafeApprovalBundle ? (
+        <>
+          Confirm (Approve&nbsp;{<TokenSymbol token={trade?.inputAmount?.currency.wrapped} length={6} />}&nbsp;and Swap)
+        </>
+      ) : undefined,
+    [isSafeApprovalBundle, trade?.inputAmount?.currency.wrapped]
+  )
```

Comment:
The actual change, plus line `91` where `buttonText` is added to component props